### PR TITLE
feat: use supervisor to manage raiden process

### PR DESCRIPTION
**Pull request by ghost**
This commit adds supervisor to raiden-simnet image in order to ensure
that the process gets restarted automatically when an unexpected crash
happens. By default supervisord will attempt to restart 3 times before
giving up.

If the changes in this PR look good I'll add similar changes to testnet/mainnet images.

### CODE REVIEW

File: images/raiden-simnet/entrypoint.sh [hunk 1]
```diff
@@ -16,4 +16,26 @@ else
     addr=`cat addr.txt`
 fi
 
-exec python -m raiden --address $addr $@
+RESTART_DELAY_SECONDS=15
+sleep $RESTART_DELAY_SECONDS
```
Comment (michael1011) [inline images/raiden-simnet/entrypoint.sh]: Doesn't this also cause the container to sleep for 15 seconds if it is started for the first time?

Comment (erkarl) [reply images/raiden-simnet/entrypoint.sh]: It does. Currently raiden crashes if geth isn't allowed some time to become responsive. Raiden was automatically restarted, but it immediately crashed again. Supervisor's default is to give up after 3 failed attempts.

Should we call it `$START_DELAY` instead?

Comment (kilrau) [reply images/raiden-simnet/entrypoint.sh]: > It does. 

And that's fine. Better play safe.

> Should we call it $START_DELAY instead?

Yep, that's clearer.

**Review by michael1011 — COMMENTED** [review]
The first raiden start still failed for me with:
```
michael@pc:~$ docker logs testnet_raiden_1 
Waiting for the GETH testnet account
Waiting for the GETH testnet account
Raiden config exists already
Welcome to Raiden, version 0.100.5a1.dev159+g03ac468b6!
----------------------------------------------------------------------
| This is an Alpha version of experimental open source software      |
| released as a test version under an MIT license and may contain    |
| errors and/or bugs. No guarantee or representations whatsoever is  |
| made regarding its suitability (or its use) for any purpose or     |
| regarding its compliance with any applicable laws and regulations. |
| Use of the software is at your own risk and discretion and by      |
| using the software you acknowledge that you have read this         |
| disclaimer, understand its contents, assume all risk related       |
| thereto and hereby release, waive, discharge and covenant not to   |
| sue Brainbot Labs Establishment or any officers, employees or      |
| affiliates from and for any direct or indirect liability resulting |
| from the use of the software as permissible by applicable laws and |
| regulations.                                                       |
|                                                                    |
| Privacy Warning: Please be aware, that by using the Raiden Client, |
| among others, your Ethereum address, channels, channel deposits,   |
| settlements and the Ethereum address of your channel counterparty  |
| will be stored on the Ethereum chain, i.e. on servers of Ethereum  |
| node operators and ergo are to a certain extent publicly available.|
| The same might also be stored on systems of parties running Raiden |
| nodes connected to the same token network. Data present in the     |
| Ethereum chain is very unlikely to be able to be changed, removed  |
| or deleted from the public arena.                                  |
|                                                                    |
| Also be aware, that data on individual Raiden token transfers will |
| be made available via the Matrix protocol to the recipient,        |
| intermediating nodes of a specific transfer as well as to the      |
| Matrix server operators.                                           |
----------------------------------------------------------------------
2019-09-30 13:33:43.598382 [info     ] Starting Raiden                [raiden.ui.runners] architecture=x86_64 distribution=source python_implementation=CPython python_version=3.7.4 raiden=0.100.5a1.dev159+g03ac468b6 raiden_db_version=23 system=Linux 64bit 5.0.0-29-generic
Raiden is running in production mode
Checking if the ethereum node is synchronized
Cannot use https://ropsten.etherscan.io/api?module=proxy&action=eth_blockNumber. Request failed
Falling back to eth_sync api.
Traceback (most recent call last):
  File "/opt/venv/lib/python3.7/site-packages/raiden/ui/startup.py", line 151, in setup_proxies_or_exit
    token_network_registry = blockchain_service.token_network_registry(registered_address)
  File "/opt/venv/lib/python3.7/site-packages/raiden/network/blockchain_service.py", line 158, in token_network_registry
    blockchain_service=self,
  File "/opt/venv/lib/python3.7/site-packages/raiden/network/proxies/token_network_registry.py", line 59, in __init__
    contract_manager.get_runtime_hexcode(CONTRACT_TOKEN_NETWORK_REGISTRY)
  File "/opt/venv/lib/python3.7/site-packages/raiden/network/rpc/client.py", line 175, in check_address_has_code
    contract_name, to_checksum_address(address)
raiden.exceptions.AddressWithoutCode: [TokenNetworkRegistry]Address 0xb9051B2Bb540297cfCeEB8C2F5D1461153B05e9F does not contain code

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.7/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/local/lib/python3.7/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/opt/venv/lib/python3.7/site-packages/raiden/__main__.py", line 17, in <module>
    main()
  File "/opt/venv/lib/python3.7/site-packages/raiden/__main__.py", line 13, in main
    run(auto_envvar_prefix="RAIDEN")  # pylint: disable=no-value-for-parameter
  File "/opt/venv/lib/python3.7/site-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/opt/venv/lib/python3.7/site-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/opt/venv/lib/python3.7/site-packages/click/core.py", line 1114, in invoke
    return Command.invoke(self, ctx)
  File "/opt/venv/lib/python3.7/site-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/opt/venv/lib/python3.7/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/opt/venv/lib/python3.7/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/opt/venv/lib/python3.7/site-packages/raiden/ui/cli.py", line 514, in run
    app = runner.run()
  File "/opt/venv/lib/python3.7/site-packages/raiden/ui/runners.py", line 246, in run
    return self._start_services()
  File "/opt/venv/lib/python3.7/site-packages/raiden/ui/runners.py", line 95, in _start_services
    app_ = run_app(**self._options)
  File "/opt/venv/lib/python3.7/site-packages/raiden/ui/app.py", line 235, in run_app
    pathfinding_service_address=pathfinding_service_address,
  File "/opt/venv/lib/python3.7/site-packages/raiden/ui/startup.py", line 156, in setup_proxies_or_exit
    "token network registry", Address(tokennetwork_registry_contract_address)
  File "/opt/venv/lib/python3.7/site-packages/raiden/ui/startup.py", line 89, in handle_contract_no_code
    hex_addr = to_checksum_address(address)
  File "/opt/venv/lib/python3.7/site-packages/eth_utils/address.py", line 109, in to_checksum_address
    norm_address = to_normalized_address(value)
  File "/opt/venv/lib/python3.7/site-packages/eth_utils/address.py", line 55, in to_normalized_address
    hex_address = hexstr_if_str(to_hex, value).lower()
  File "/opt/venv/lib/python3.7/site-packages/eth_utils/conversions.py", line 154, in hexstr_if_str
    return to_type(hexstr_or_primitive)
  File "/opt/venv/lib/python3.7/site-packages/eth_utils/decorators.py", line 68, in wrapper
    _assert_one_val(*args, **kwargs)
  File "/opt/venv/lib/python3.7/site-packages/eth_utils/decorators.py", line 33, in _assert_one_val
    "Instead, values were: %r, %r" % (args, kwargs)
TypeError: Exactly one of the passed values can be specified. Instead, values were: (None,), {}
```

But I guess that is not related to this PR?

**Comment by erkarl** [comment]
@michael1011 the changes in this PR only affect simnet environment. I'm waiting for a review from @reliveyy so that I can apply the same changes for testnet/mainnet.

**Review by michael1011 — APPROVED** [review]
> the changes in this PR only affect simnet environment

Oh, I overlooked that. The PR works fine on simnet :+1:

**Pull request merged**